bundles/org.openhab.binding.automower/src/main/java/org/openhab/binding/automower/internal/rest/api/HusqvarnaApi.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.automower.internal.rest.api;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.HttpClient;
+import org.openhab.binding.automower.internal.rest.exceptions.AutomowerCommunicationException;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * Base class for the Husqvarna apis
+ *
+ * @author Markus Pfleger - Initial contribution
+ */
+@NonNullByDefault
+public abstract class HusqvarnaApi {
+
+    private final HttpClient httpClient;
+    protected final Gson gson;
+
+    protected HusqvarnaApi(HttpClient httpClient) {
+        this.httpClient = httpClient;
+
+        GsonBuilder gsonBuilder = new GsonBuilder();
+        gson = gsonBuilder.create();
+    }
+
+    protected abstract String getBaseUrl();
+
+    protected HttpClient getHttpClient() throws AutomowerCommunicationException {
+        if (!httpClient.isRunning()) {
+            try {
+                httpClient.start();
+            } catch (Exception e) {
+                throw new AutomowerCommunicationException("Exception while trying to start http client", e);
+            }
+        }
```

Comment:
Did you encounter a case where the HTTP client is not running?

Reply:
I was not sure about the status of the httpclient that I get. To avoid starting it multiple times, or having it not started at all, I added the isRunning check. I am not sure if it is already started or not as I get the httpClient from the framework...

Reply:
I think the framework takes care of starting it. I didn't see it in other bindings. But I can't give a definite answer. Let's wait for the opinion of the second reviewer.

Reply:
I will try without starting it explicitly. I think you are probably right, would be strange to get an httpclient injected that would need to be started. If the binding would be required to start it would also be required to stop it, I guess the client will be shared between bindings...

Reply:
Ok, seems to work, I completely removed this block